### ISSUE TITLE
Is macOS supported?

### ISSUE THREAD

**Issue by RealPortalPlayer** [darlinghq/darling#1337]
Don't know if this was asked before, but, does this work on macOS?
Before you say it, yes, I get the irony of trying to run a macOS emulation layer on macOS, similar to running Wine on Windows.
However, macOS Catalina, and above, doesn't support 32-bit applications, while this (at least from what I've seen) does.
I'm asking if this could eventually be used to bring 32-bit support back to macOS? Similar to being able to run 16-bit applications via Wine. (At least, last time I checked it did)

**Comment by trungnt2910** [comment]
It would be very hard, as Darling is not cross-platform by design and uses a lot of Linux-specific non-portable APIs.

**Comment by bugaevc** [comment]
No, neither macOS, nor other systems (the BSDs...) are supported; we're only doing Linux. It could be possible to support other kernels (especially now that there's no longer a kernel module) provided that:
* they have a stable kernel ABI (which Darwin does _not_, which is why Darling can exist in the first place...);
* support the features that we need (which as @trungnt2910 says is non-trivial, we're using stuff like overlayfs and namespaces and pidfd);
* someone is interested enough to implement & support this.
